friends/src/Components/Friend.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from 'react'
+import styled from 'styled-components'
+import Input from './Input'
+
+const EnableEdit = styled.button`
+    height: 100%;
+    border-top-right-radius: 10px;
+    border-bottom-right-radius: 10px;
+    border: 0;
+    background: red;
+    width: 70px;
+`
+
+const Submit = styled.button`
+    height: 100%;
+    border-top-right-radius: 10px;
+    border-bottom-right-radius: 10px;
+    border: 0;
+    background: limegreen;
+    width: 70px;
+`
+
+class Friend extends Component {
+    constructor(props) {
+        super()
+        this.props = props
+        this.state = {
+            name: props.friend.name,
+            age: props.friend.age,
+            email: props.friend.email,
+            disabled: true
+        }
+    }
+
+    editEnabler = (e) => {
+        e.preventDefault()
+        this.setState({
+            disabled: false
+        })
+    }
+
+    changeHandler = (e) => {
+        this.setState({
+            ...this.state,
+            [e.target.name]: e.target.value
+        })
+    }
+
+    submitEdit = (e) => {
+        e.preventDefault()
+
+    }
+
+    render() {
+        return (
+            <li>
+                <form onSubmit={event => this.props.submitEdit(event)}>
+                    <ul>
+                        <li><input name="name" value={this.state.name} disabled={this.state.disabled} onChange={(event) => this.changeHandler(event)} /></li>
+                        <li><input name="age" value={this.state.age} disabled={this.state.disabled} onChange={(event) => this.changeHandler(event)} /></li>
+                        <li><input name="email" value={this.state.email} disabled={this.state.disabled} onChange={(event) => this.changeHandler(event)} /></li>
```

Comment:
The same thing about the unnecessary steps apply here and above. Also, the lines being too long applies too! This also all look very similar. Why not use a function to make these? For example, here's one you could stick at the top of your render function before your return:
```javascript
const makeInput = name => (
  <li>
    <input
      name={name}
      value={this.state[name]}
      disabled={this.state.disabled}
      onChange={this.eventHandler}
    />
  </li>
);
```

Reply:
and the usage, for example:
```javascript
<ul>
  {[ 'name', 'email', 'age' ].map(makeInput)}
</ul>
```